url_shortener/views/handlers.py

```diff
@@ -1,60 +1,332 @@
 """This module contains various examples on how to implement an endpoint"""
 import http.client as httplib
+import re
+import random
+import string
 from url_shortener.logic import Logic
+from url_shortener.dto import User
 
 from pyramid.request import Request
 from pyramid.response import Response
+from pyramid.httpexceptions import HTTPFound
 
+EMAIL_REGEX = re.compile(r"[^@]+@[^@]+\.[^@]+")
+domain_name = 'localhost:6543/'
 
-def protected_resource_read_example(request: Request) -> Response:
-    key = request.matchdict['key']
+
+def get_auth_token(request: Request) -> str:
+    """ Strips request header to get the Auth id only. """
+    return request.headers['Authorization'].replace('Bearer ', '')
+
+
+# Old name of the method: protected_resource_write_example
+def create_user(request: Request) -> Response:
+    """ Creates user if all conditions are met.
+        CONDITIONS
+            - If non-null data entered.
+            - If account does not exist.
+            - If username does not exits.
+            - If username is longer than 4 chars
+            - If email address is valid.
+        Returns a response.
+     """
     logic: Logic = request.registry.logic
-    value = logic.get_example(key)
+    try:
+        new_user = User(email=request.json_body.get('email'),
+                        username=request.json_body.get('username'),
+                        password=request.json_body.get('password'),
+                        id=logic.get_user_count())
+    except Exception as e:
+        print(e)
+        return Response(status=httplib.BAD_REQUEST, json_body={
+            'status': 'error',
+            'description': 'Request body is missing.'
+        })
+
+    # Conditions.
```

Comment:
You can use [`trafaret`](https://trafaret.readthedocs.io/en/latest/intro.html) library included in the project for validation.